sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java

```diff
@@ -212,7 +211,7 @@ String id() {
             this.id = id;
             this.secret = secret;
 
-            if (this.baseUri == null || this.id == null || this.secret == null) {
+            if (this.baseUri == null || CoreUtils.isNullOrEmpty(this.id) || this.secret == null) {
```

Comment:
Why is id treated differently than baseUri or secret, won't all have issues if null or empty?

Reply:
id' is type of String, could be null or empty
but baseUri is URL type, `private final URL baseUri;`.